files/build_templates/sonic_debian_extension.j2

```diff
@@ -646,12 +646,19 @@ sudo mkdir -p $FILESYSTEM_ROOT/$PLATFORM_DIR/common
 sudo cp {{ deb }} $FILESYSTEM_ROOT/$PLATFORM_DIR/common/
 sudo ln -sf "../common/{{ debfilename }}" "$FILESYSTEM_ROOT/$PLATFORM_DIR/{{dev}}/{{ debfilename }}"
 for f in $(find $FILESYSTEM_ROOT/var/cache/apt/archives -name "*.deb"); do
-    sudo mv $f $FILESYSTEM_ROOT/$PLATFORM_DIR/{{dev}}/
+    sudo mv $f $FILESYSTEM_ROOT/$PLATFORM_DIR/common/
+    sudo ln -sf "../common/$(basename $f)" "$FILESYSTEM_ROOT/$PLATFORM_DIR/{{dev}}/$(basename $f)"
 done
 
 sudo dpkg --root=$FILESYSTEM_ROOT -P {{ debname }}
 
 {% endfor %}
+# create a trivial apt repo if any of the debs have dependencies, including between lazy debs
+if [ $(for f in $FILESYSTEM_ROOT/$PLATFORM_DIR/common/*.deb; do \
+           sudo dpkg -I $f | grep "Depends:\|Pre-Depends:"; done | wc -l) -gt 0 ]; then
+    sudo dpkg-scanpackages $FILESYSTEM_ROOT/$PLATFORM_DIR/common | \
+        sudo gzip | sudo tee $FILESYSTEM_ROOT/$PLATFORM_DIR/common/Packages.gz > /dev/null
+fi
```

Comment:
Can we list out all the packages and the order that are required for a specific platform? Can We simple installed the in sequence?
For instance, we have 3 packages, A.deb, B.deb, and C.deb, A-depends->B-depends->C.
Packages.list
```
C.deb B.deb A.deb
```
We can reduce the complex logic to install the packages, line 287 to 300, make thing simple.
It is only one of the suggestions, not sure if it works for you.

Reply:
@xumia Just looking for predictable packaging from the vendor side. I will try refactoring as you suggest.  apt-get has different orders for the unpack and configure steps.  Unpack remains a lexicographical sort except when PreDepends= is used.  Configure is a topological sort by Depends= .  Both ordered lists are in the stdout of apt install -s (simulate) but screen scraping is sketchy business....maybe there is another way to capture.   But in the end, apt is there and its logic is maintained and tested by several distros.

Reply:
@xumia Since apt wraps dpkg I wrote a wrapper for dpkg to see if we could get the list reliably.   This methodology seems to work, but I would like your input before I invest much more.

/usr/bin/dpkg wrapper:

```
#!/bin/bash
A="$@"
#echo "dpkg $A" >> /var/log/apt.trace
if [[ $A == *"unpack"* ]]; then
    if [[ $A == *"recursive"* ]]; then 
        # apt decide to rename the packages in temp space
        ls ${A##* } | cut -d '-' -f 2-  >> /var/log/dpkg.order
    else
        # apt listed the debs on the command line
        for word in $A; do
            if [[ $word == *".deb" ]]; then
                echo ${word##*/} >> /var/log/dpkg.order
            fi
        done
    fi	
fi
/usr/bin/dpkg.bin $A
```

The installer becomes (an appropriate filename when done):

`dpkg -I $(cat /var/log/dpkg.order)`

Of course there is all the other setup moved to the build template and a bunch of testing.  Thoughts?

Reply:
How about changing the installation script, unpack the packages, and then configure them.
```
dpkg --force-all --unpack *
dpkg --configure -a
```

Test 1: Failed
```
docker run --rm -it debian:bullseye bash
Inside of the docker container:
cd ~
apt-get update
apt-get download $(apt-cache depends --recurse --no-recommends --no-suggests --no-conflicts --no-breaks --no-replaces --no-enhances python3-dev | grep "^\w")
dpkg -i *

```
Error messages as below:
```
Processing triggers for libc-bin (2.31-13+deb11u4) ...
Errors were encountered while processing:
 python3-minimal_3.9.2-3_amd64.deb
 python3_3.9.2-3_amd64.deb
 python3-dev
 python3-distutils
 python3-lib2to3
```

Test 2: succeeded
```
docker run --rm -it debian:bullseye bash
Inside of the docker container:
cd ~
apt-get update
apt-get download $(apt-cache depends --recurse --no-recommends --no-suggests --no-conflicts --no-breaks --no-replaces --no-enhances python3-dev | grep "^\w")
dpkg -i *
```

Only some warning messages:
```
dpkg: regarding python3-minimal_3.9.2-3_amd64.deb containing python3-minimal, pre-dependency problem:
 python3-minimal pre-depends on python3.9-minimal (>= 3.9.2-0~)
  python3.9-minimal is not installed.

dpkg: warning: ignoring pre-dependency problem!


```

Reply:
Thats pretty much what apt does under the covers, except apt sorts the dependent packages in order first. Apt either renames the debs with numeric prefixes in temp space or supplies them on the command line in order.  The order sets the configure order.  The problem we have is that different debs deliver different systemd services and we need the postint scripts [which start them] to be run in order.

Reply:
After reviewing with @xumia The late binding in the installer appears to carry the lowest risk and liability. Syncing the fork to rerun the checks.